ssg/components.py

```diff
@@ -5,7 +5,21 @@
 
 import ssg.yaml
 
-
+try:
+    from functools import lru_cache
+except ImportError:  # Viva la python 2.7
+    from functools import wraps
+    def lru_cache(max_size, user_function):
+        cache = {}
+        @wraps(user_function)
+        def wrapper(*args):
+            key = tuple(args)
+            if key not in cache:
+                cache[key] = user_function(*args)
+            return cache[key]
+        return wrapper
+
+@lru_cache(max_size=16) # A speed up cache. We have been opening these same 150 files 300 times per a product build.
```

Comment:
```suggestion
@lru_cache(maxsize=16) # A speed up cache. We have been opening these same 150 files 300 times per a product build.
```

Reply:
Thanks. Will do it manually as there are two occurences to fix.